Cubical/Data/Nat/Primes/Lemmas.agda

```diff
@@ -0,0 +1,321 @@
+
+<1→0 : ∀ n → n < 1 → n ≡ 0
+<1→0 n n<1 with (<-split n<1)
+... | inr n=0 = n=0
+... | inl n<0 = ⊥.rec (¬-<-zero n<0)
+
+<<-asym : ∀ {a b} → a < b → ¬ b < a
+<<-asym {a} {b} a<b b<a = ¬n<n (<-trans a<b b<a)
+
+suc<-elim : ∀ {a} {b} → suc a < suc b → a < b
+suc<-elim {a} {b} (n , n+sa=b) = n , injSuc (subst (λ x → x ≡ suc b) (+-suc n (suc a)) n+sa=b)
+
+suc< : ∀ {a} {b} → a < b → suc a < suc b
+suc< {a} {b} (n , n+sa=b) = n , subst (λ x → x ≡ suc b) (sym (+-suc n (suc a))) (cong suc n+sa=b)
+
+add< : ∀ {a} {b} d → a < b → d + a < d + b
+add< zero a<b = a<b
+add< (suc d) a<b = suc< (add< d a<b)
+
+mult< : ∀ k a b → a < b → a · suc k < b · suc k
+mult< k a zero a<b = ⊥.rec (¬-<-zero a<b)
+mult< k zero (suc b) a<b = z<suc (k + b · suc k)
+mult< k (suc a) (suc b) a<b = suc< (add< k (mult< k a b (suc<-elim a<b)))
+
+lem1 : ∀ c k d n → c < k → k · (suc d) < n → c · (suc d) < n
+lem1 c k d n c<k ksd<n = <-trans (mult< d c k c<k) ksd<n
+
+lem2 : ∀ c k d n → k < c → n < d + k · d → n < d + c · d
+lem2 c k 0 n k<c n<k0 = ⊥.rec (¬-<-zero (subst (λ x → n < x) (sym (0≡m·0 k)) n<k0))
+lem2 c k d@(suc d-1) n k<c n<d+kd = <-trans n<d+kd (add< d (mult< d-1 k c k<c))
+
+sa<n : ∀ a n → suc a < n → a < n
+sa<n a n (k , k+ssa=n) = k + 1 , subst (λ x → x ≡ n) (+-assoc k 1 (suc a)) k+ssa=n
+
++a<n : ∀ k a n → k + a < n → a < n
++a<n zero a n a<n = a<n
++a<n (suc k) a n sk+a<n = +a<n k a n (sa<n (k + a) n sk+a<n)
+
+add-equations : ∀ {a} {b} {c} {d} → a ≡ b → c ≡ d → a + c ≡ b + d
+add-equations {b = b} {c = c} a=b c=d = cong (_+ c) a=b ∙ cong (b +_) c=d
+
+a·pos<n : ∀ a b n → a · suc b < n → a < n
+a·pos<n zero b n z<sucn = z<sucn
+a·pos<n (suc a) b zero asb<n = ⊥.rec (¬-<-zero asb<n)
+a·pos<n (suc a) b (suc n) asb<n = suc< (a·pos<n a b n (+a<n b (a · suc b) n (suc<-elim asb<n)))
+
+inj-·0< : ∀ {k} a b → 0 < k → k · a ≡ k · b → a ≡ b
+inj-·0< {zero} _ _ 0<0 _ = ⊥.rec (¬n<n 0<0)
+inj-·0< {suc k-1} a b 0<k = inj-sm· {m = k-1}
+
+n≤n·pos : ∀ n k → 0 < k → n ≤ n · k
+n≤n·pos n zero 0<0 = ⊥.rec (¬n<n 0<0)
+n≤n·pos n (suc k) _ = subst (λ x → n ≤ x) (sym (·-comm n (suc k))) ≤SumLeft
+
+1<·1<=3< : ∀ {a b} → 1 < a → 1 < b → 3 < a · b
+1<·1<=3< {0} {0} 1<a 1<b = ⊥.rec (1<→¬=0 0 1<a refl)
+1<·1<=3< {0} {suc b} 1<a 1<b = ⊥.rec (1<→¬=0 0 1<a refl)
+1<·1<=3< {suc a} {0} 1<a 1<b = ⊥.rec (1<→¬=0 0 1<b refl)
+1<·1<=3< {1} {suc b} 1<a 1<b = ⊥.rec (¬n<n 1<a)
+1<·1<=3< {suc (suc a)} {1} 1<a 1<b = ⊥.rec (¬n<n 1<b)
+1<·1<=3< {suc (suc a)} {suc (suc b)} 1<a 1<b =
+    suc< (suc< (<≤-trans (suc< (z<suc (b + a · suc (suc b)))) (≤SumRight {k = b})))
+
+PropFac< : ∀ a b n → 1 < b → 0 < n → a · b ≡ n → a < n
+PropFac< zero _ _ _ 0<n _ = 0<n
+PropFac< (suc a) zero n 1<b 0<n ab=n = ⊥.rec (¬-<-zero 1<b)
+PropFac< (suc a) (suc zero) n 1<b 0<n ab=n = ⊥.rec (¬n<n 1<b)
+PropFac< (suc a) (suc (suc b)) zero _ 0<n ab=n = ⊥.rec (¬n<n 0<n)
+PropFac< (suc a) (suc (suc b)) (suc n) _ _ ab=n = lem a b n ab=n where
+    lem : ∀ a b n → suc (suc (b + a · suc (suc b))) ≡ suc n → suc a < suc n
+    lem a b n p =
```

Comment:
Nice! I played a bit with your notation, you can also do this (if you like):
<img width="1113" height="433" alt="image" src="https://github.com/user-attachments/assets/ebe3692d-8a7a-4427-9ecc-8a3ca13eddd8" />
I replaced `→` with `⇒` just because I associate the latter more with "implication", but I guess that depends on how you grew up - certainly no strong opinion there.